src/lib/types.hpp

```diff
@@ -188,7 +188,7 @@ PredicateCondition flip_predicate_condition(const PredicateCondition predicate_c
 // ">" becomes "<=" etc.
 PredicateCondition inverse_predicate_condition(const PredicateCondition predicate_condition);
 
-enum class JoinMode { Inner, Left, Right, Outer, Cross, Semi, Anti };
+enum class JoinMode { Inner, Left, Right, FullOuter, Cross, Semi, AntiDiscardNulls, AntiRetainNulls };
```

Comment:
This needs a comment. The other types are common knowledge, but we should doc the NULL behavior here. Also, we should doc why the same does not apply to semi.